sale_import_base/components/importer_sale_channel.py

```diff
@@ -0,0 +1,202 @@
+#  Copyright (c) Akretion 2020
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from marshmallow_objects import ValidationError as MarshmallowValidationError
+
+from odoo.exceptions import ValidationError
+
+from odoo.addons.component.core import Component
+
+
+class ImporterSaleChannel(Component):
+    _inherit = "processor"
+    _name = "importer.sale.channel"
+    _apply_on = ["sale.order"]
+    _usage = "json_import"
+
+    def run(self, raw_data):
+        """
+        :param raw_data: json-like string
+        :return: generated sale order
+        """
+        # we need this step because the chunk data is stored as
+        # a string (so we can edit it)
+        try:
+            so_datamodel_load = self.env.datamodels["sale.order"].load_json(raw_data)
+        except MarshmallowValidationError:
+            raise ValidationError(MarshmallowValidationError)
+        data = so_datamodel_load.dump()
+        so_vals = self._prepare_vals(data)
+        new_sale_order = self.env["sale.order"].create(so_vals)
+        self._finalize(new_sale_order, data)
+        return new_sale_order
+
+    def _prepare_vals(self, data):
+        partner = self._process_partner(data)
+        address_invoice, address_shipping = self._process_addresses(
+            partner, data["address_invoicing"], data["address_shipping"]
+        )
+        so_vals = {
+            "partner_id": partner.id,
+            "partner_invoice_id": address_invoice.id,
+            "partner_shipping_id": address_shipping.id,
+            "si_amount_total": data["amount"]["amount_total"],
+            "si_amount_untaxed": data["amount"]["amount_untaxed"],
+            "si_amount_tax": data["amount"]["amount_tax"],
+            "order_line": [
+                (0, 0, self._prepare_sale_line(line)) for line in data["lines"]
+            ],
+            "sale_channel_id": self.collection.record_id,
+        }
+        result = self._execute_onchanges(so_vals)
+        return result
+
+    def _process_partner(self, data):
+        partner = self._find_partner(data)
+        vals = self._extract_address_info(data["address_customer"])
+        if partner:
+            partner.write(vals)
```

Comment:
Here we systematically commit all the address info that is in the JSON import to the partner in database. Is this OK ?

Reply:
by that I mean, we always just use a write with the info, without checking anything else